src/main/java/org/mitre/synthea/engine/ExpressedSymptom.java

```diff
@@ -0,0 +1,181 @@
+package org.mitre.synthea.engine;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ExpressedSymptom implements Cloneable, Serializable {
+  
+  private static final long serialVersionUID = 4322116644425686800L;
+  
+  // this class contains basic info regarding an expressed symptoms.
+  // such as the cause and the associated value
+  public class SymptomInfo implements Cloneable, Serializable {    
+    private static final long serialVersionUID = 4322116644425686801L;
+    // what is the cause of the symptom
+    public String cause; 
+    // what is the value associated to that symptom
+    public Integer value;
+    
+    public SymptomInfo(String cause, Integer value) {
+      this.cause = cause;
+      this.value = value;
+    }
+    
+    public SymptomInfo clone() {
+      return new SymptomInfo(this.cause, this.value);
+    }
+    
+    public String getCause() {
+      return cause;
+    }
+
+    public Integer getValue() {
+      return value;
+    }    
+  }
+
+  // this class encapsulates module-based infos regarding an expressed symptoms.
+  public class SymptomSource implements Cloneable, Serializable {    
+    private static final long serialVersionUID = 4322116644425686802L;
+    
+    ExpressedSymptom symptom = ExpressedSymptom.this;
+    // From which module the expressed symptom was set    
+    private String source;
+    // what is the status from a given expressed symptom from a given module
+    private boolean status;
+    // when the expressed was last updated from the a given module
+    private Long lastUpdateTime;
```

Comment:
Instead of compute it each time we need to get the currentValue of a symtom, I just keep track it each time the value of a symtom is updated. I removed the setLastUpdateTime() method for to deal with inconsistency issues.
also this field is only updated internally when the value of a symptom is updated.